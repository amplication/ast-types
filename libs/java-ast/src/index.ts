// Core exports
export * from "./lib/core/AstNode";
export * from "./lib/core/Writer";

// AST nodes exports
export * from "./lib/ast/Access";
export * from "./lib/ast/Annotation";
export * from "./lib/ast/Class";
export * from "./lib/ast/ClassReference";
export * from "./lib/ast/CodeBlock";
export * from "./lib/ast/Enum";
export * from "./lib/ast/Field";
export * from "./lib/ast/GenericType";
export * from "./lib/ast/Import";
export * from "./lib/ast/Interface";
export * from "./lib/ast/Method";
export * from "./lib/ast/Parameter";
export * from "./lib/ast/Type";

// Utils exports
// (Add utility functions as needed)
