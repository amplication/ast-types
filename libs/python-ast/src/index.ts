export * from "./lib/ast";
export * from "./lib/core";
