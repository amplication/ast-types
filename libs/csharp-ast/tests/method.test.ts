import { CsharpSupport, MethodType, CsharpSupport as csharp } from "../src/";

describe("method", () => {
  it("basic", () => {
    const myMethod = csharp.method({
      access: "public",
      name: "MyVoidMethod",
      isAsync: true,
      return_: csharp.Types.genericReference(
        CsharpSupport.genericClassReference({
          reference: csharp.classReference({
            name: "ActionResult",
            namespace: "System.Threading.ActionResult",
          }),
          innerType: CsharpSupport.Types.list(
            CsharpSupport.Types.reference(
              CsharpSupport.classReference({
                name: `myDto`,
                namespace: `MyService.APIs.Dtos`,
              })
            )
          ),
        })
      ),
      parameters: [
        csharp.parameter({
          name: "simpleParameter",
          type: csharp.Types.integer(),
        }),
        csharp.parameter({
          name: "listParameter",
          type: csharp.Types.list(csharp.Types.string()),
        }),
        csharp.parameter({
          name: "optionalParameter",
          type: csharp.Types.optional(csharp.Types.string()),
        }),
      ],
      body: csharp.codeblock({
        code: 'Console.WriteLine(@"Received these: {simpleParameter} {listParameter} {optionalParameter}");',
      }),
    });

    expect(myMethod.toString()).toMatchSnapshot();
  });

  it("extension method with parameters", () => {
    const myMethod = csharp.method({
      access: "public",
      name: "MyExtensionMethod",
      isAsync: true,
      type: MethodType.STATIC,
      extensionParameter: csharp.parameter({
        name: "extensionParameter",
        type: csharp.Types.integer(),
      }),
      parameters: [
        csharp.parameter({
          name: "simpleParameter",
          type: csharp.Types.integer(),
        }),
        csharp.parameter({
          name: "listParameter",
          type: csharp.Types.list(csharp.Types.string()),
        }),
        csharp.parameter({
          name: "optionalParameter",
          type: csharp.Types.optional(csharp.Types.string()),
        }),
      ],
    });

    expect(myMethod.toString()).toMatchSnapshot();
  });

  it("extension method without parameters", () => {
    const myMethod = csharp.method({
      access: "public",
      name: "MyExtensionMethod",
      isAsync: true,
      type: MethodType.STATIC,
      extensionParameter: csharp.parameter({
        name: "extensionParameter",
        type: csharp.Types.integer(),
      }),
      parameters: [],
    });

    expect(myMethod.toString()).toMatchSnapshot();
  });
});
