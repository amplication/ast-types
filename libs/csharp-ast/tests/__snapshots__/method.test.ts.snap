// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`method basic 1`] = `
"using System.Threading.ActionResult;
using MyService.APIs.Dtos;

public async Task<ActionResult<List<myDto>>> MyVoidMethod(int simpleParameter, List<string> listParameter, string? optionalParameter) {
    Console.WriteLine(@"Received these: {simpleParameter} {listParameter} {optionalParameter}");}
"
`;

exports[`method extension method with parameters 1`] = `
"public static async Task MyExtensionMethod(this int extensionParameter, int simpleParameter, List<string> listParameter, string? optionalParameter) {
}
"
`;

exports[`method extension method without parameters 1`] = `
"public static async Task MyExtensionMethod(this int extensionParameter) {
}
"
`;
